OpenRA.Mods.Common/Widgets/Logic/Ingame/Hotkeys/EditorQuickSaveHotkeyLogic.cs

```diff
@@ -0,0 +1,77 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2022 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using OpenRA.FileSystem;
+using OpenRA.Mods.Common.Lint;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Widgets;
+
+namespace OpenRA.Mods.Common.Widgets.Logic.Ingame
+{
+	[ChromeLogicArgsHotkeys("EditorQuickSaveKey")]
+	public class EditorQuickSaveHotkeyLogic : SingleHotkeyBaseLogic
+	{
+		readonly World world;
+		readonly ModData modData;
+
+		[TranslationReference]
+		static readonly string SaveCurrentMap = "save-current-map";
+
+		[ObjectCreator.UseCtor]
+		public EditorQuickSaveHotkeyLogic(Widget widget, ModData modData, World world, Dictionary<string, MiniYaml> logicArgs)
+			: base(widget, modData, "QuickSaveKey", "GLOBAL_KEYHANDLER", logicArgs)
+		{
+			this.world = world;
+			this.modData = modData;
+		}
+
+		protected override bool OnHotkeyActivated(KeyInput keyInput)
+		{
+			var map = world.Map;
+			var editorActorLayer = world.WorldActor.Trait<EditorActorLayer>();
+			Action<string> saveMap = (string combinedPath) =>
```

Comment:
This is mostly just a duplicate of the code inside `SaveMapLogic.cs`, we should try to reuse code. That would also have the benefit of adding a transient notification there